tests/unit/test_repository.py

```diff
@@ -623,55 +627,96 @@ def test_create_repository_client(
     assert isinstance(returned_client, repository.Client)
 
 
-def test_repository_summary_modified(repository_client):
+@pytest.mark.parametrize(
+    "folder",
+    [
+        pytest.param(DOCUMENTATION_FOLDER_NAME, id="docs folder only"),
+        pytest.param(None, id="all path"),
+    ],
+)
+def test_repository_summary_modified(repository_client, folder: Optional[str]):
     """
     arrange: given repository with a modified file with respect to the head
     act: when get_summary is called
     assert: The DiffSummary object represents the modified file.
     """
-    (repository_client.base_path / "file-1.txt").write_text("My first version")
-    repository_client.update_branch("file-1 commit", force=False, push=False)
+    base_path = repository_client.base_path / folder if folder else repository_client.base_path
+    if folder:
+        base_path.mkdir()
```

Comment:
I think this can be done through an indirect fixture, which will allow cleanup once the test is done

Reply:
Not sure I got your point, I was referencing using the `indirect` parameter to pass the parameter through a fixture
See: https://docs.pytest.org/en/7.1.x/example/parametrize.html#apply-indirect-on-particular-arguments

It's helpful when you have an attribute coming from parametrize that you initialize and want to cleanup after execution of the test

Reply:
Oh, I see. Never used it before. I have added that now. Thanks for the tip